### DISCUSSION TITLE
Edit/move bookmark without having to remove it

### DISCUSSION THREAD

**Discussion by pitommasi** [alefragnani/vscode-bookmarks#476] [Q&A] [answered]
As the title says: I saved a file externally and the bookmarks are now all messed up.
Apparently, from the interface I can only edit the label and not the position.
As a fallback I'm editing them in the .vscdb file but it's not a quick task.

Is there a quicker way to edit the bookmark to restore its location?

**Comment by alefragnani** [comment, marked as answer]
Hi @pitommasi,

Unfortunately, no, you can't edit/update the position.

If you were using `bookmarks.saveBookmarksInProject` setting as `true`, the bookmarks would be saved in `.vscode/bookmarks.json`, inside the folder you opened, instead of VS Code internals, and would be easier to update.

I remember last year someone tried to _change the setting_ to make the bookmarks to be save locally, but the bookmarks disapppeared. I have fixed this a few months later in (https://github.com/alefragnani/vscode-bookmarks/issues/242 ) and it seems to be working just fine. You could give it a try (maybe try with a _less important_ folder first, and if everything works, do the same on the real folder. Better yet, you are already touching the SQLite database, so simply make a backup, to avoid any loss.

Btw, feel free to open an issue for that idea. Maybe other users have the same request.

Hope this helps

**Reply by pitommasi** [reply]
Thanks, I'll give that setting a try and will open an issue with the feature request.

**Comment by apollokit** [comment]
Question on this. I'd love to be able to edit my list of bookmarks, and I don't mind doing so by directly editing the bookmarks.json file. Is there any way to edit the order with a json file when the `bookmarks.saveBookmarksInProject` option is false? Or do I need to edit the SQLite database directly?

Context: My understanding is the only way to get a .json file I can directly edit is using the `bookmarks.saveBookmarksInProject` option. I'm not really a fan of using  that option, because I have a multiroot project and as such when I use the "list all bookmarks" command I always have to select which root to use, in the first pop-up (I'd like it if there were no root selection, and it just displayed a flat list of all bookmarks across all roots, immediately)

**Reply by alefragnani** [reply]
Hi @apollokit ,

For now, you would need to edit SQLite directly, but I would ask you, why do you need to edit the _order_ of bookmarks? I mean, there are a few _sort related_ feature requests still open  (https://github.com/alefragnani/vscode-bookmarks/issues?q=is%3Aissue+is%3Aopen+sort+). None of the sorts will work for you?

Be aware that even if you edit the bookmarks.json` file/SQLite, it won’t update the bookmarks automatically. You will need to reload the window.

Also, about the multi-root scenarios, you could upvote #417 , which the idea is to remove that folder picker.

Hope this helps
